### ISSUE TITLE
New Rule: no-space-after-hash-on-atx-style-heading

### ISSUE THREAD

**Issue by SwetaTanwar** [eslint/markdown#370]
### Rule details

This rule will warn when spaces are missing after the hash characters in an atx style heading:

### What type of rule is this?

Warns about a potential problem

### Example code

```markdown
#Heading 1

##Heading 2
```

### Participation

- [x] I am willing to submit a pull request to implement this rule.

### Additional comments

Prior At: [Rule](https://github.com/DavidAnson/markdownlint/blob/main/doc/Rules.md#md018---no-space-after-hash-on-atx-style-heading)

**Comment by lumirlumir** [comment]
Thanks for mentioning me, @Pixel998. I mentioned this rule in another comment, but I didn’t raise an issue about it—so I’m fine with whoever decides to take it on.

It’s a good rule to implement, especially since it’s already covered in [markdownlint](https://github.com/DavidAnson/markdownlint/blob/main/doc/Rules.md#md018---no-space-after-hash-on-atx-style-heading).

That said, please keep in mind that this rule could be tricky to implement due to potential edge cases.

I’m in favor of creating this rule, but I’d like to hear what the rest of the team thinks before moving forward. @eslint/eslint-team

**Comment by SwetaTanwar** [comment]
I will pick this up and submit a PR

**Comment by snitin315** [comment]
👍🏻 for `no-missing-space-atx`, I think we should also have this as a recommendation.
